### PR TITLE
Webkit2gtk

### PR DESCRIPTION
**Pull request by Ludea**

### CODE REVIEW

**Comment by Ludea** [comment]
build script seems to not clone repo

**Comment by xeffyr** [comment]
@Ludea Because submodule is not updated.

**Comment by xeffyr** [comment]
Rebase on latest commit from `master`.

**Comment by xeffyr** [comment]
Rebase fixed the issue.

`ERROR: No files in package` means that sources do not have known build system files: ./configure, CMakeLists.txt or Makefile. Packaging is not straightforward and there is a reason why webkit is still not packaged for Termux.

**Comment by Ludea** [comment]
There is a Cmakelist.txt
https://svn.webkit.org/repository/webkit/releases/WebKitGTK/webkit-2.31.1/CMakeLists.txt

**Comment by Ludea** [comment]
`ERROR: Package webkit depends on non-existing package "libhypen"
` whereas I rebase

**Comment by xeffyr** [comment]
Probably because you have a custom submodule update in your fork, commit https://github.com/termux/x11-packages/pull/264/commits/600444ac2a8194f39277ab2372b693c8b1d9e0c8. You need to remove it.

Just found that package `libhypen` is incorrectly named: should be `libhyphen`. So this will be needed to be rebased again.

**Comment by Ludea** [comment]
I rebase again and update submodule, still issue with another package

**Comment by Grimler91** [comment]
@Ludea the termux-packages submodule in [your branch](https://github.com/Ludea/x11-packages/tree/webkit) is not updated, it points to termux-packages @ 5b7e66e, which was 8 days ago. If you rebase against x11-packages, and remove the commits where you have updated the termux-packages submodule then it will probably work

**Comment by Grimler91** [comment]
To update the submodule (though in this case you should just rebase onto x11-packages master and not touch the submodule) you can run

```sh
git submodule update --init --remote
git add termux-packages
git commit -m "Update termux-packages submodule"
```

File: packages/webkit/SharedMemoryUnix.cpp.patch [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+             fileDescriptor = shm_open(tempName.data(), O_CREAT | O_RDWR, S_IRUSR | S_IWUSR);
+-        } while (fileDescriptor == -1 && errno == EINTR);
++        } while (fileDescriptor == -1 && errno == EINTR);*/
+     }
+
+-    if (fileDescriptor != -1)
+-        shm_unlink(tempName.data());
++   /* if (fileDescriptor != -1)
++        shm_unlink(tempName.data());*/
+
```
Comment (xeffyr) [inline packages/webkit/SharedMemoryUnix.cpp.patch:20]: See https://github.com/termux/x11-packages/blob/master/packages/i3/posix-shm-implementation.patch for shm_open / shm_unlink example.

**Review by xeffyr — CHANGES_REQUESTED** [review]

File: packages/webkit/SharedMemoryUnix.cpp.patch (hunk 1 above)
Comment (Ludea) [reply packages/webkit/SharedMemoryUnix.cpp.patch]: Well done

**Review by xeffyr — APPROVED** [review]
Ok, merging it.

**Comment by Ludea** [comment]
Thanks a lot ;)

**Pull request merged**